### DISCUSSION TITLE
export p12 in macOS Monterey

### DISCUSSION THREAD

**Discussion by cocoaoa** [airsdk/Adobe-Runtime-Support#2063] [General]
Hi, before Monterey, you could export the p12 certificate from Keyaccess.app exporting the certificate and the private key together:
![959ab736-dc2c-4096-946d-4e27fb162bcf](https://user-images.githubusercontent.com/111195509/184504357-7d816e79-2a96-4e4e-8192-e9d983d0e3ba.jpg)

But now, I can't find the private key:
![b54a9424-f598-4f15-a443-12ee7c400826](https://user-images.githubusercontent.com/111195509/184504344-1220df68-1848-4d30-ae76-4b9d83e09178.jpg)

any hekp? 

I need this in order to export the app from Animate and install it in my device before sending to apps stores

**Comment by ajwfrost** [comment]
I still see a private key for the certificates where I'd imported both together.. but you're looking at two different places here: the first screenshot shows your login items and the "my certificates" bit which is where I would expect to find the signing certificates that you've added and for which you have the key. But the lower screenshot is the system items and just all certificates..

So perhaps you need to get this again from the Apple Developer portal or have this imported somehow into your local certificate store? I'm afraid I don't recall how I did this, but I can see that I do have a number of valid certificates plus private keys under my "login" keychain.

thanks

**Reply by cocoaoa** [reply]
Hi!

Thanks for answering.

I got no answer at the developer forums at Apple…

The 2 developers I know they use Air still using Mojave…. :(

I'm aware about "my certificates" screenshot but there is no difference between both  "my certificates" and "login items" I mean, I can't see private key in none of them.

OK, so you can see both private key and certificate together in Monterey as in early systems? then I don't know how to do that because I had to repeat all the process in Mojave and an old iPhone with iOS 14… there the process is straight, but some how, the same steps in Monterey doesn't give me both items together…

**Reply by toysonlocation** [reply]
It has the appearance of a missing key. Normally you can expand the certificate and see a subordinate key. Can you download the certificate again from the portal? Maybe click on "Inicio de session" ("login" in my screenshot)?

![image](https://user-images.githubusercontent.com/39204807/184976575-4b9f0c32-1e4e-4452-9158-6aecad5d473d.png)

**Reply by cocoaoa** [reply]
Hi, thanks. I’ve already tried that a few times, but the private key never appears

**Reply by toysonlocation** [reply]
Does this still happen if you create a new certificate from scratch, starting with the request and going through the whole process on the Appple developer site?...

<img width="721" alt="image" src="https://user-images.githubusercontent.com/39204807/185008680-cb685952-7fc2-4dcc-bfbe-2eb2ddd368df.png">

**Reply by marchbold** [reply]
As @toysonlocation said, this looks like the behaviour when you haven't created the request from the same machine. In those cases you'll only have part of the certificate if you download it from the developer console and won't be able to export the complete p12. If you have your original p12 file you should be able to add that to the keychain, otherwise restarting the process should work.
